release_notes/ocp-4-12-release-notes.adoc

```diff
@@ -2232,6 +2232,25 @@ sourceStrategy:
 
 * Previously, although the reference `PolicyGenTemplate` for `group-du-sno-ranGen.yaml` includes two `StorageClass`  entries, the generated policy included only one. With this update, the generated policy now includes both policies. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2049306[*BZ#2049306*]).
 
+* Previously, attempting to install an {sno} cluster with two Day 2 workers failed when using `ignitionEndpoint`.
+After the initial spoke cluster installation, the `AgentClusterInstall` never reached the `adding-hosts` state and remained in the `installed` state. Additionally, the admin `kubeconfig` secret was not generated on the hub cluster, so the `kubeapi` of the spoke cluster could not be accessed.
+With this update, installing an {sno} cluster with two Day 2 workers succeeds when using `ignitionEndpoint`.
+The `AgentClusterInstall` now reaches the `adding-hosts` state and the admin `kubeconfig` secret is generated on the hub cluster, allowing access to the `kubeapi` of the spoke cluster.
+
+* Previously, when deploying a spoke cluster via Assisted Service (Infra Operator) in a disconnected environment and referencing a public {product-title} image to install, the assisted service pod would not recognize the `ImageContentSourcePo` (ICSP) object and fail to pull the image.
+Setting the `ICSP` in the `AgentClusterInstall` as an `install-config` override and relying on the mirror `configmap` also resulted in the same error.
+With this update, the assisted service pod recognizes the `ICSP` and pulls the image.
+There is no longer an error when setting the `ICSP` in the `AgentClusterInstall` as an `install-config` override when using the mirror `configmap`.
+
+* Previously, when using the assisted-service to complete the Agent installation, the agent stage remained in a Rebooting state. With this update, the assisted-service can be used to complete the Agent installation.
+
+* Previously, there were no validation checks performed on the installation of an {sno} cluster with `platformType == VSphere`, causing failed installation. With this update, validation blocks the installation from starting and indicates that the {sno} cluster is incompatible with the VSphere platform.
+
+* Previously, when adding Day 2 workers to an {sno} cluster, certificate signing requests (CSRs) were not auto-approved and did not automatically join the cluster. With this update, CSRs are auto-approved and nodes join the cluster automatically.
+
+* Previously, when adding a Day 2 worker with an illegal role (not primary, worker, auto-assign), the agent resource reported `specsynced` as successful instead of reporting an "Illegal role" error, and the `agentclusterinstall` displayed a back-end error for `specsynced`.
```

Comment:
```suggestion
* Previously, when adding a Day 2 worker with an illegal role (not primary, worker, auto-assign), the agent resource reported `specsynced` as successful instead of reporting an "Illegal role" error, and the `agentclusterinstall` displayed a back-end error for `specsynced`. With this update, the agent resource reports the back-end error, and agent validations fail when `specsynced` reports "Illegal role defined".
```

Reply:
Updated as suggested.